tests/util/test_struct_stream.py

```diff
@@ -245,7 +248,7 @@ def test_uint32_from_bytes(self) -> None:
         assert uint32(b"273") == 273
 
     def test_struct_stream_cannot_be_instantiated_directly(self) -> None:
-        with pytest.raises(ValueError, match="does not fit"):
+        with pytest.raises(AttributeError, match="object has no attribute"):
```

Comment:
I was initially thinking about just dropping this test because im not sure why its a problem to instantiate a `StructStream` if you wish to do so.

Now with removing the default values of the members it's also not possible to instantiate, like before but yeah with a different error. Because it doesn't have members assigned. For me it feels like thats actually a better way of restricting instantiation as we had before, no? 

The way before didn't work anymore because you could instantiate a `StructStream(0)` with the changed condition in the `__init__` because 

```python
if not (self.MINIMUM <= self <= self.MAXIMUM):
    raise ValueError(f"Value {self} does not fit into {type(self).__name__}")
```

doesn't trigger here with the new `<=0` which was `<0` before.  

At the end i removed the default values then to trigger it this way because i didn't want to add some extra condition into the `__init__` like `self.MINIMUM != self.MAXIMUM` or so.

Reply:
Agreed that we don't want an extra conditional in the init slowing down all other cases (which is presumably _all_ cases).  0 could become -1 though, similar to all the other off-by-one shifts.  If we return to this pattern, it should get a comment.  I should have had one before.

As to going out of our way to restrict instantiation...  maybe not?  I mean, I certainly regularly advocate for the consenting adults perspective.  Limits could be `-math.inf` and `math.inf` I guess...  Though I think that the methods would simply not work without sensible class attributes configuring them and I don't know that such limits exist without prescribing the default to be of a particular size or making it auto-sized with extra work and code or...  So, maybe this is less about consenting adults and more about 'this class simply will not function in any sensible way so no, you do not get to make an instance and we explicitly make it so you can not'.